greenlet recipe build fail

### Logs

```
In file included from ./slp_platformselect.h:47:
./platform/switch_aarch64_gcc.h:62:51: warning: value size does not match register size specified by the constraint and modifier [-Wasm-operand-widths]
           __asm__ volatile ("mov %0, #0" : "=r" (err));
                                                  ^
./platform/switch_aarch64_gcc.h:62:35: note: use constraint modifier "w"
           __asm__ volatile ("mov %0, #0" : "=r" (err));
                                  ^~
                                  %w0
greenlet.c:538:10: error: no member named 'use_tracing' in 'struct _ts'
        tstate->use_tracing = 0;
        ~~~~~~  ^
greenlet.c:541:10: error: no member named 'use_tracing' in 'struct _ts'
        tstate->use_tracing = (tstate->tracing <= 0 &&
        ~~~~~~  ^
greenlet.c:823:13: warning: 'PyEval_CallObjectWithKeywords' is deprecated [-Wdeprecated-declarations]
                        result = PyEval_CallObjectWithKeywords(
                                 ^
/home/bobf/ex/regression/.buildozer/android/platform/build-arm64-v8a/build/other_builds/python3/arm64-v8a__ndk_target_21/python3/Include/ceval.h:17:1: note: 'PyEval_CallObjectWithKeywords' has been explicitly marked deprecated here
Py_DEPRECATED(3.9) PyAPI_FUNC(PyObject *) PyEval_CallObjectWithKeywords(
^
/home/bobf/ex/regression/.buildozer/android/platform/build-arm64-v8a/build/other_builds/python3/arm64-v8a__ndk_target_21/python3/Include/pyport.h:513:54: note: expanded from macro 'Py_DEPRECATED'
#define Py_DEPRECATED(VERSION_UNUSED) __attribute__((__deprecated__))
                                                     ^
greenlet.c:1009:19: error: expression is not assignable
                Py_REFCNT(self) = 1;
                ~~~~~~~~~~~~~~~ ^
greenlet.c:1038:7: error: expression is not assignable
                if (--Py_REFCNT(self) != 0) {
                    ^ ~~~~~~~~~~~~~~~
greenlet.c:1042:20: error: expression is not assignable
                        Py_REFCNT(self) = refcnt;
                        ~~~~~~~~~~~~~~~ ^
greenlet.c:1046:4: error: use of undeclared identifier '_Py_DEC_REFTOTAL'
                        _Py_DEC_REFTOTAL;
                        ^
2 warnings and 6 errors generated.
```

--- a/pythonforandroid/recipes/greenlet/__init__.py
+++ b/pythonforandroid/recipes/greenlet/__init__.py
@@ -1,8 +1,8 @@
-from pythonforandroid.recipe import CompiledComponentsPythonRecipe
+from pythonforandroid.recipe import PyProjectRecipe
 
 
-class GreenletRecipe(CompiledComponentsPythonRecipe):
-    version = '0.4.15'
+class GreenletRecipe(PyProjectRecipe):
+    version = '3.1.1'
     url = 'https://pypi.python.org/packages/source/g/greenlet/greenlet-{version}.tar.gz'
     depends = ['setuptools']
     call_hostpython_via_targetpython = False

